src/nemos/basis/basis.py

```diff
@@ -0,0 +1,1478 @@
+    >>> from nemos.basis import RaisedCosineLinearEval
+    >>> n_basis_funcs = 5
+    >>> raised_cosine_basis = RaisedCosineLinearEval(n_basis_funcs)
+    >>> sample_points = np.random.randn(100)
+    >>> # convolve the basis
+    >>> features = raised_cosine_basis.compute_features(sample_points)
+    """
+
+    def __init__(
+        self,
+        n_basis_funcs: int,
+        width: float = 2.0,
+        bounds: Optional[Tuple[float, float]] = None,
+        label: Optional[str] = "RaisedCosineLinearEval",
+    ):
+        EvalBasisMixin.__init__(self, bounds=bounds)
+        RaisedCosineBasisLinear.__init__(
+            self,
+            n_basis_funcs,
+            width=width,
+            mode="eval",
+            label=label,
+        )
+
+    @add_docstring("evaluate_on_grid", RaisedCosineBasisLinear)
+    def evaluate_on_grid(self, n_samples: int) -> Tuple[NDArray, NDArray]:
+        """
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from nemos.basis import RaisedCosineLinearEval
+        >>> n_basis_funcs = 5
+        >>> decay_rates = np.array([0.01, 0.02, 0.03, 0.04, 0.05]) # sample decay rates
+        >>> window_size=10
+        >>> ortho_basis = RaisedCosineLinearEval(n_basis_funcs)
+        >>> sample_points, basis_values = ortho_basis.evaluate_on_grid(100)
+
+        """
+        return super().evaluate_on_grid(n_samples)
+
+    @add_docstring("_compute_features", EvalBasisMixin)
+    def compute_features(self, xi: ArrayLike) -> FeatureMatrix:
+        """
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from nemos.basis import RaisedCosineLinearEval
+
+        >>> # Generate data
+        >>> num_samples = 1000
+        >>> X = np.random.normal(size=(num_samples, ))  # raw time series
+        >>> basis = RaisedCosineLinearEval(10)
+        >>> features = basis.compute_features(X)  # basis transformed time series
+        >>> features.shape
+        (1000, 10)
+
+        """
+        return super().compute_features(xi)
+
+    @add_docstring("split_by_feature", RaisedCosineBasisLinear)
+    def split_by_feature(
+        self,
+        x: NDArray,
+        axis: int = 1,
+    ):
+        r"""
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from nemos.basis import RaisedCosineLinearEval
+        >>> from nemos.glm import GLM
+        >>> basis = RaisedCosineLinearEval(n_basis_funcs=6, label="one_input")
+        >>> X = basis.compute_features(np.random.randn(20,))
+        >>> split_features_multi = basis.split_by_feature(X, axis=1)
+        >>> for feature, sub_dict in split_features_multi.items():
+        ...        print(f"{feature}, shape {sub_dict.shape}")
+        one_input, shape (20, 1, 6)
+
+        """
+        return super().split_by_feature(x, axis=axis)
+
+
+class RaisedCosineLinearConv(
+    ConvBasisMixin, RaisedCosineBasisLinear, BasisTransformerMixin
```

Comment:
why is BasisTransformerMixin inherited here? doesn't it get it through RaisedCosineBasisLinear (which gets it from Basis)?

Reply:
thanks, it's a left over, I was trying the mixin class